game/view/actor/attr.lua

```diff
@@ -0,0 +1,148 @@
+
+local RES     = require 'resources'
+local RANDOM  = require 'common.random'
+local APT     = require 'domain.definitions.aptitude'
+local COLORS  = require 'domain.definitions.colors'
+local FONT    = require 'view.helpers.font'
+local PLAYSFX = require 'helpers.playsfx'
+
+local abs  = math.abs
+local min  = math.min
+local max  = math.max
+local pi   = math.pi
+local fmod = math.fmod
+
+local coresume = coroutine.resume
+local cocreate = coroutine.create
+local coyield  = coroutine.yield
+
+local delta = love.timer.getDelta
+
+local _dt
+local _barwidth
+local _font
+local _rot = 0
+local _particles = {}
+local _states = {}
+
+local function _newParticleSource()
+  local pixel = RES.loadTexture('pixel')
+  local particles = love.graphics.newParticleSystem(pixel, 128)
+  particles:setParticleLifetime(.75)
+  particles:setSizeVariation(0)
+  particles:setLinearDamping(6)
+  particles:setSpeed(256)
+  particles:setSpread(2*pi)
+  particles:setColors(COLORS.NEUTRAL, COLORS.TRANSP)
+  particles:setSizes(4)
+  particles:setEmissionArea('ellipse', 0, 0, 0, false)
+  particles:setTangentialAcceleration(-512)
+  return particles
+end
+
+local function _renderAttribute(data)
```

Comment:
Why can't the coroutine just receive three parameters (`actor`, `attrname`, and `particles`)?

Reply:
Also, the `particles` is redundant, isn't it? You can get it from `actor` and `attrname`